Source/Server/Server.c

```diff
@@ -317,6 +317,54 @@ void stop_server(void)
     server.running = 0;
 }
 
+static void _set_sigaction(const int sig, const struct sigaction *sigact, const char *sig_name)
+{
+    /*
+    GCC recommends to not replace existing sigaction structs
+    if they specify SIG_IGN
+    */
+    struct sigaction prev_act;
+    sigaction(sig, NULL, &prev_act);
+    if(prev_act.sa_handler == SIG_IGN)
+    {
+        // signal should be ignored - so let's not tamper with it
+        return;
+    } 
+
+    // sigaction returns 0 if success, -1 if error
+    if(sigaction(sig, sigact, NULL))
+    {
+        LOG_ERROR("Unable to set sigaction for %s", sig_name);
+    }
+}
+
+static void _handle_signal(const int sig)
+{
+    switch(sig)
+    {
+        case SIGINT:
+            handle_sigint();
+            break;
+        case SIGTERM:
+            stop_server();
+            break;
+        default:
+            LOG_ERROR("Received unhandled signal %d", signal);
+            break;
```

Comment:
should unhandled signals (`default` branch) run `stop_server` after that log? or just continue as if nothing happened.